src/tree.rs

```diff
@@ -256,18 +266,43 @@ impl<Pane> Tree<Pane> {
             preview_rect: None,
         };
 
-        let rect = ui.available_rect_before_wrap();
-
+        let mut rect = ui.available_rect_before_wrap();
+        if self.height.is_finite() {
+            rect.set_height(self.height);
+        }
+        if self.width.is_finite() {
+            rect.set_width(self.width);
+        }
         if let Some(root) = self.root {
             self.tiles.layout_tile(ui.style(), behavior, rect, root);
 
             self.tile_ui(behavior, &mut drop_context, ui, root);
         }
 
         self.preview_dragged_tile(behavior, &drop_context, ui);
+        ui.allocate_space(rect.size());
+    }
+
+    /// Sets the exact height that can be used by the tree.
+    ///
+    /// determines the height that will be used by the tree component, by default, the tree occupies all the available space in the parent container
+    pub fn set_height(&mut self, height: f32) {
+        if height.is_sign_positive() && height.is_finite() {
+            self.height = height;
+        } else {
+            self.height = f32::INFINITY;
+        }
+    }
 
-        // Allocate the used space in the parent Ui:
-        ui.advance_cursor_after_rect(rect);
```

Comment:
Bad merge! `allocate_space` should be removed and `advance_cursor_after_rect` re-added